tnt-application/src/main/java/org/dcsa/tnt/controller/EventController.java

```diff
@@ -54,60 +71,112 @@ public List<EventTO> findEvents(
+        .eventCreatedDateTime(queryParameterParser.parseCustomQueryParameter(queryParams, "eventCreatedDateTime", OffsetDateTime::parse))
+        .eventDateTime(queryParameterParser.parseCustomQueryParameter(queryParams, "eventDateTime", OffsetDateTime::parse))
         .eventType(toEnumList(eventType, EventType.class))
-        .shipmentEventTypeCode(shipmentEventTypeCode)
-        .carrierBookingReference(carrierBookingReference)
-        .transportDocumentReference(transportDocumentReference)
-        .transportEventTypeCode(transportEventTypeCode)
-        .transportCallID(transportCallID)
+        .shipmentEventTypeCode(toEnumList(shipmentEventTypeCode, ShipmentEventTypeCode.class))
+        .transportEventTypeCode(toEnumList(transportEventTypeCode, TransportEventTypeCode.class))
+        .equipmentEventTypeCode(toEnumList(equipmentEventTypeCode, EquipmentEventTypeCode.class))
+        .documentTypeCode(toEnumList(documentTypeCode, DocumentTypeCode.class))
+        .documentReference(documentReference)
+        .transportCallReference(transportCallReference)
+        .carrierExportVoyageNumber(carrierExportVoyageNumber)
+        .universalExportVoyageReference(universalExportVoyageReference)
+        .universalServiceReference(universalServiceReference)
+        .UNLocationCode(UNLocationCode)
         .vesselIMONumber(vesselIMONumber)
-        .carrierVoyageNumber(carrierVoyageNumber)
         .carrierServiceCode(carrierServiceCode)
-        .equipmentEventTypeCode(equipmentEventTypeCode)
         .equipmentReference(equipmentReference)
-      .build());
+      .build(),
+      eventMapper::toDTO);
 
     paginator.setPageHeaders(request, response, cursor, result);
     return result.content();
   }
+
+  private final SortBy[] defaultSort = new SortBy[] { new SortBy(Sort.Direction.ASC, EventCache_.EVENT_CREATED_DATE_TIME) };
+  private final Set<String> sortableFields = Set.of(EventCache_.EVENT_CREATED_DATE_TIME, EventCache_.EVENT_DATE_TIME);
+
+  private Cursor.SortBy[] parseSort(String sort) {
+    if (sort == null) {
+      return defaultSort;
+    }
+
+    return Arrays.stream(sort.split(","))
+      .map(String::trim)
+      .map(sortField -> {
+        String[] fieldAndDirection = sortField.split(":");
+        String actualSortField = fieldAndDirection[0];
+        Direction direction = Direction.ASC;
+
+        if (!sortableFields.contains(actualSortField)) {
+          throw ConcreteRequestErrorMessageException.invalidInput("Cannot sort on '" + actualSortField + "'");
+        }
+
+        if (fieldAndDirection.length == 2) {
+          direction = Direction.fromOptionalString(fieldAndDirection[1])
+            .orElseThrow(() -> ConcreteRequestErrorMessageException.invalidInput(
+              "'" + fieldAndDirection[1] + "' is not a valid direction"));
+        } else if (fieldAndDirection.length > 2) {
+          throw ConcreteRequestErrorMessageException.invalidInput("'" + sortField + "' is not valid");
```

Comment:
I think `ConcreteRequestErrorMessageException.invalidQuery("sort", ...)` would be better (more specific)